clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -504,6 +504,27 @@ public boolean refreshCommittedOffsetsIfNeeded(final long timeoutMs) {
         return true;
     }
 
+    /**
+     * Refresh the committed offsets for provided partitions.
+     * 
+     * @param startMs   The time in which the operation starts
+     * @param timeout   The maximum allowable duration of the method
+     * @throws TimeoutException if committed offsets cannot be retrieved within set amount of time
+     */
+    public void refreshCommittedOffsetsIfNeeded(long startMs, long timeout) {
```

Comment:
Similar to the previous comment. Why do we need new methods that duplicate all this logic?